### PR TITLE
Add bcq auth token command and global --host flag

### PR DESCRIPTION
**Pull request by jeremy**
## Summary

- Add `bcq auth token` command to print OAuth token to stdout for use with other tools
- Add global `--host` flag with smart protocol detection for multi-environment workflows
- Improve auth error messaging with origin URLs and underlying error details

## Usage

```bash
# Export token for curl or other tools
export BASECAMP_TOKEN=$(bcq auth token)

# Get token for different environments
bcq --host localhost:3000 auth token
bcq --host staging.example.com auth token

# Use stored OAuth credentials (ignore BASECAMP_TOKEN env)
bcq auth token --stored
```

## Host normalization

The `--host` flag intelligently detects the protocol:
- `localhost`, `127.0.0.1`, `[::1]` → `http://`
- All other hostnames → `https://`
- Explicit URLs used as-is

## Test plan

- [x] `make` passes (lint, vet, tests)
- [x] `bats e2e/auth.bats` passes (14 tests)
- [x] Manual testing of `bcq auth token` with/without BASECAMP_TOKEN
- [x] Manual testing of `--host` flag with different environments

### CODE REVIEW

File: internal/auth/auth.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (m *Manager) AccessToken(ctx context.Context) (string, error) {
 	origin := config.NormalizeBaseURL(m.cfg.BaseURL)
 	creds, err := m.store.Load(origin)
 	if err != nil {
-		return "", output.ErrAuth("Not authenticated")
+		return "", output.ErrAuth(fmt.Sprintf("Not authenticated for %s: %v", origin, err))
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/auth/auth.go:73]: Intentional design choice - we explicitly want to surface the underlying error (keyring access failure, file permissions, etc.) to help users debug auth issues. The detailed error is more actionable than a generic message. For example, "secret not found in keyring" tells the user to re-login, while "keyring denied" might indicate a system permission issue.

File: internal/auth/auth.go [hunk 2]
```diff
@@ -83,8 +83,45 @@ func (m *Manager) AccessToken(ctx context.Context) (string, error) {
...
+// refreshing if needed. Unlike AccessToken, this ignores the BASECAMP_TOKEN
+// environment variable and always uses stored OAuth credentials.
+func (m *Manager) StoredAccessToken(ctx context.Context) (string, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	origin := config.NormalizeBaseURL(m.cfg.BaseURL)
+	creds, err := m.store.Load(origin)
+	if err != nil {
+		return "", output.ErrAuth(fmt.Sprintf("No stored credentials for %s: %v", origin, err))
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/auth/auth.go:107]: Same rationale as above - detailed errors help users diagnose auth issues. The underlying error distinguishes between "credentials not found" (need to login) vs "keyring access denied" (system issue) vs "JSON parse error" (corrupted credentials file).

File: internal/auth/auth.go [hunk 3]
```diff
@@ -114,7 +151,7 @@ func (m *Manager) Refresh(ctx context.Context) error {
 	origin := config.NormalizeBaseURL(m.cfg.BaseURL)
 	creds, err := m.store.Load(origin)
 	if err != nil {
-		return output.ErrAuth("Not authenticated")
+		return output.ErrAuth(fmt.Sprintf("Not authenticated for %s: %v", origin, err))
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/auth/auth.go:154]: Same rationale - detailed errors are intentional for debugging auth issues.

File: internal/cli/root.go [hunk 4]
```diff
@@ -196,6 +198,41 @@ func Execute() {
...
+	}
+	if host == "127.0.0.1" || strings.HasPrefix(host, "127.0.0.1:") {
+		return true
+	}
+	// IPv6 loopback
+	if host == "[::1]" || strings.HasPrefix(host, "[::1]:") {
+		return true
+	}
+	return false
+}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cli/root.go:234]: Fixed - added comprehensive unit tests in internal/cli/root_test.go covering:
- Various localhost formats (localhost, 127.0.0.1, [::1], ::1)
- Bare hostnames vs full URLs
- Edge cases like localhost.example.com
- IPv6 with and without brackets
- Empty strings

File: internal/cli/root.go [hunk 5]
```diff
@@ -73,16 +76,15 @@ func NewRootCmd() *cobra.Command {
 	cmd.PersistentFlags().StringVarP(&flags.Project, "project", "p", "", "Project ID or name")
 	cmd.PersistentFlags().StringVarP(&flags.Account, "account", "a", "", "Account ID")
 	cmd.PersistentFlags().StringVar(&flags.Todolist, "todolist", "", "Todolist ID or name")
-	cmd.PersistentFlags().StringVar(&flags.BaseURL, "base-url", "", "Basecamp API base URL")
+	cmd.PersistentFlags().StringVar(&flags.Host, "host", "", "Basecamp host (e.g., localhost:3000, staging.example.com)")
+	cmd.PersistentFlags().StringVar(&flags.Host, "base-url", "", "Basecamp API base URL (deprecated: use --host)")
+	_ = cmd.PersistentFlags().MarkHidden("base-url")
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cli/root.go:81]: This is intentional - last-flag-wins is standard CLI behavior (same as how other flags work). The --base-url is hidden/deprecated and only exists for backward compatibility with existing scripts. Users should use --host going forward. Adding a warning would add complexity for a deprecated flag that most users will never see.

File: internal/cli/root.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/cli/root.go]: Fixed - added support for bare `::1` (without brackets). Now handles both `[::1]:3000` and `::1` formats. Added unit test coverage for this case.
